src/api/handler/handler.js

```diff
@@ -1,17 +1,25 @@
-import {axiosNasaClient} from "../../clients/index.js";
+import {nasaAsteroidService} from "../../services/index.js";
 import {getPreviousPeriod, parseDate} from "./dateUtils.js";
 
+const parseQueryParams = (query) => {
+    return {
+        date: query.date,
+        countOnly: Boolean(query.count),
+        dangerousOnly: Boolean(query["were-dangerous-meteors"]),
+    }
+}
+
 export const handler = async (req, res) => {
-    const dateQuery = req.query.date;
-    const {errorMessage, date} = parseDate(dateQuery);
+    const {date, countOnly, dangerousOnly} = parseQueryParams(req.query);
+    const {errorMessage} = parseDate(date);
     if (errorMessage) {
         res.status(400).send(errorMessage);
         return;
```

Comment:
you can throw exception here and then handle it later in your middleware pipeline
This approach will allow you to handle all errors uniformly

Reply:
Thank you! I moved this validation to middleware. Fixed